### PR TITLE
adding ad component, high score, more app.json details, removing comp…

### PR DESCRIPTION
**Pull request by craig1123**
…onentsWillMount as they'll be depracated, starting rate app

### CODE REVIEW

File: react-native-slide-puzzle/src/reducers/boardReducer.js [hunk 1]
```diff
@@ -32,12 +38,17 @@ function arraysEqual(arr1, arr2) {
...
 export default function (state = INITIAL_STATE, action) {
-  console.log(action)
   switch (action.type) {
     case MOVE_TILE:
       let won = false;
       if (arraysEqual(action.payload.board, state.winningArray)) {
         won = true;
+        const maybeHighScore = state.moves + 1;
+        getHighScore().then(highScore => {
+          if (maybeHighScore < parseInt(highScore)) {
```
Comment (CamWalker) [inline react-native-slide-puzzle/src/reducers/boardReducer.js]: if the potential high score is less than the current high score, then make it the high score?

File: react-native-slide-puzzle/src/containers/Board.js [hunk 2]
```diff
@@ -1,11 +1,22 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
-import { View, Text, TouchableOpacity, StyleSheet, ImageEditor, StatusBar, Dimensions } from 'react-native';
-import { ImagePicker, ScreenOrientation } from 'expo';
```
Comment (CamWalker) [inline react-native-slide-puzzle/src/containers/Board.js]: Why are we taking out screen orientation locking?

File: react-native-slide-puzzle/src/reducers/boardReducer.js (hunk 1 above)
Comment (craig1123) [reply react-native-slide-puzzle/src/reducers/boardReducer.js]: Correct. In this game, it is golf. The less moves, the better

File: react-native-slide-puzzle/src/containers/Board.js (hunk 2 above)
Comment (craig1123) [reply react-native-slide-puzzle/src/containers/Board.js]: Ahh, I took it out in the componentWillMount, but NOT in the app. We don't have to define it in our game, but in our app.json file which defines the settings of our app

File: react-native-slide-puzzle/src/reducers/boardReducer.js (hunk 1 above)
Comment (CamWalker) [reply react-native-slide-puzzle/src/reducers/boardReducer.js]: That's right. My bad

**Pull request merged**